java/code/src/com/redhat/rhn/domain/formula/FormulaFactory.java

```diff
@@ -911,6 +911,13 @@ else if (pillarExampleFileCustom.isFile()) {
     @SuppressWarnings("unchecked")
     public static Map<String, Object> disableMonitoring(Map<String, Object> formData) {
         Map<String, Object> exporters = (Map<String, Object>) formData.get("exporters");
+        // If we cannot extract the exporters from the form data the form data file might
+        // come from an older version of the prometheus-exporters formula package and we will
+        // use the example file instead
+        if (exporters == null) {
+            formData = FormulaFactory.getPillarExample(PROMETHEUS_EXPORTERS);
+            exporters = (Map<String, Object>) formData.get("exporters");
+        }
```

Comment:
This will overwrite the user's configuration with defaults.

Reply:
Yes this is intended, if the files contents cannot be read by our codebase, we override the file with the defaults. This is to prevent NPE if exporters could not be extracted see: https://github.com/SUSE/spacewalk/issues/15350

Reply:
In this case the customer already had an up-to-date `prometheus-exporters-formula` package. So if there is a migration for these files something must have gone wrong